src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs

```diff
@@ -130,34 +133,59 @@ public static LogEventLevel FormDataLoggingLevel
         /// <summary>
         /// Initializes a module and prepares it to handle requests.
         /// </summary>
-        /// <param name="context">An <see cref="T:System.Web.HttpApplication"/> that provides access to the methods, properties, and events common to all application objects within an ASP.NET application </param>
-        public void Init(HttpApplication context)
-        {
-            context.LogRequest += LogRequest;
-            context.Error += Error;
-        }
-
-        static void LogRequest(object sender, EventArgs e)
+        /// <param name="application">An <see cref="T:System.Web.HttpApplication"/> that provides access to the methods, properties, and events common to all application objects within an ASP.NET application </param>
+        public void Init(HttpApplication application)
         {
-            if (!_isEnabled) return;
-
-            var request = HttpContextCurrent.Request;
-            if (request == null)
-                return;
+            application.BeginRequest += (sender, args) =>
+            {
+                if(_isEnabled && application.Context != null)
+                {
+                    application.Context.Items[StopWatchKey] = Stopwatch.StartNew();
+                }                
+            };
 
-            Logger.Write(_requestLoggingLevel, "HTTP {Method} for {RawUrl}", request.HttpMethod, request.RawUrl);
-            if (ShouldLogRequest())
+            application.EndRequest += (sender, args) =>
             {
-                var form = request.Unvalidated.Form;
-                if (form.HasKeys())
+                if (_isEnabled && application.Context != null)
                 {
-                    var formData = form.AllKeys.SelectMany(k => (form.GetValues(k) ?? new string[0]).Select(v => new { Name = k, Value = FilterPasswords(k, v) }));
-                    Logger.Write(_formDataLoggingLevel, "Client provided {@FormData}", formData);
+                    Stopwatch stopwatch = (Stopwatch)application.Context.Items[StopWatchKey];
```

Comment:
Looking good! One other thought - if `IsEnabled` was  set to `true` during the request, `stopwatch` will be `null` here. I think it'll need an additional check.

`var stopwatch` is probably more in keeping with the style of the rest of the file.

LGTM otherwise, ready to merge after these tweaks.